node_create_transaction.go

```diff
@@ -0,0 +1,416 @@
+package hedera
+
+import (
+	"time"
+
+	"github.com/hashgraph/hedera-protobufs-go/services"
+)
+
+/*-
+ *
+ * Hedera Go SDK
+ *
+ * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/**
+ * A transaction to create a new node in the network address book.
+ * The transaction, once complete, enables a new consensus node
+ * to join the network, and requires governing council authorization.
+ * <p>
+ * This transaction body SHALL be considered a "privileged transaction".
+ * <p>
+ *
+ * - MUST be signed by the governing council.
+ * - MUST be signed by the `Key` assigned to the
+ *   `admin_key` field.
+ * - The newly created node information SHALL be added to the network address
+ *   book information in the network state.
+ * - The new entry SHALL be created in "state" but SHALL NOT participate in
+ *   network consensus and SHALL NOT be present in network "configuration"
+ *   until the next "upgrade" transaction (as noted below).
+ * - All new address book entries SHALL be added to the active network
+ *   configuration during the next `freeze` transaction with the field
+ *   `freeze_type` set to `PREPARE_UPGRADE`.
+ *
+ * ### Record Stream Effects
+ * Upon completion the newly assigned `node_id` SHALL be in the transaction
+ * receipt.
+ */
+type NodeCreateTransaction struct {
+	Transaction
+	accountID           *AccountID
+	description         string
```

Comment:
nit: I think this should be optional if we follow the proto definitions. 
```
 /**
   * A short description of the node.
   * This value, if set, MUST NOT exceed 100 bytes when encoded as UTF-8.
   * This field is OPTIONAL.
   */
```

Reply:
Does this mean to make them as pointers?

Reply:
Yes, but it is not important. Will check things further for the topic.